### ISSUE TITLE
[V1][Help Wanted] Porting missing sampling parameters to V1

### ISSUE THREAD

**Issue by WoosukKwon** [vllm-project/vllm#13058]
### Anything you want to discuss about vllm.

To switch the engine from V0 to V1, we need to comprehensively support the sampling parameters in https://github.com/vllm-project/vllm/blob/main/vllm/sampling_params.py

While most of the key parameters are already supported, some of them are missing:

TODO (help wanted):
- [x] `n` (parallel sampling) #10980  @afeldman-nm 
- [x] `guided_decoding` (structured decoding) #12388  @aarnphm 
- [x] `logit_bias` #13079 @houseroad 
- [x] `min_p` #13191 @AoyuQC
- [ ] `bad_words` (originally implemented via logits processor) #13376 @22quinn 
- [x] `allowed_token_ids` (originally implemented via logits processor) #13210 @houseroad 

Parameters that will not be supported in V1:
* best_of
* logits_processors

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by 22quinn** [comment]
Seems like good first issues - @WoosukKwon I'd like to take a stab at `bad_words`

**Comment by akeshet** [comment]
```
Parameters that will not be supported in V1:

...
logits_processors
```

Is this to say that support for logits_processors is going to be permanently dropped in V1, or is it just out of scope for this particular GH issue?

**Comment by WoosukKwon** [comment]
@22quinn Thanks for volunteering! Could you please submit a PR by EoW?

**Comment by WoosukKwon** [comment]
@akeshet 

We plan to re-design the API for that. We will probably not allow per-request logits processor (because this is too complex and slow). We are exploring other options.

Please refer to this comment from @njhill: https://github.com/vllm-project/vllm/pull/12688#issuecomment-2631679428

**Comment by njhill** [comment]
We'll still have a logits processor plugin mechanism (not per request) but the interface will be different. I think that @AlpinDale was working on a proposal for this.

It might be nice for some of these sampling parameters to be implemented via that same abstraction but assuming it won't be ready before we need them we can plan to retrofit where it makes sense.

**Comment by maliknaik16** [comment]
@22quinn, are you still working on this task? If not, I'd be happy to take it up! Let me know what you think. 

cc @WoosukKwon

**Comment by WoosukKwon** [comment]
@maliknaik16 Please feel free to take it!

@22quinn Let us know if you already have the PR.

**Comment by 22quinn** [comment]
@maliknaik16 @WoosukKwon I'm on it, PR will be up soon

**Comment by WoosukKwon** [comment]
@22quinn Oh great. Thanks!

**Comment by lk-chen** [comment]
As #13376 is merged, I think all TODOs are done, and this issue can be closed?
